submissions/recipe_oriented_factorio_life/recipe_oriented_factorio_life.md

```diff
@@ -0,0 +1,75 @@
+
+### Module design
+
+Let's start with the simplest example, the iron gear wheel, a recipe with a single ingredient and a single product. A logistic train stop at the bottom of the grid is requesting 8000 iron plates, and a logistic train stop at the top of the grid is providing items (i.e. gears) from the connected warehouse.
+
+![Gear module](figure/rofl_gear_module.jpg)
+_Simply gear module. Raw material station is below, product station is above_
+
+Since the inputs and outputs within this module are fulfilled by trains, our modification and upgrade of the production line within the module will not affect other modules.
+
+![Alt gear module](figure/rofl_gear_module_2.jpg)
+_Gear module with beacons. No change of stations, only the production line between stations._
+
+Next, let‘s look at an advanced circuit module with three types of raw materials. By setting the logistic train stop requesting multiple ingredients, we were able to unload all of our raw materials at a single station. At this point all the raw materials are mixed in one warehouse， we then use filter miniloaders (similar to filter inserters) to separate each material onto a separate belt to supply the assembly machine.
+
+![Advanced circuit module](figure/advanced_circuit_module.jpg)
+_Advanced circuit module. Use a single station to obtain raw materials._
+
+The feedstock of recipes that produce/consume fluids can also be transported in a fluid wagon, such as oil processing.
+
+![Oil refinery module](figure/oil_refinery_module.jpg)
+_Oil refinery module. Note that by controlling the pump via the LTN signal allows different fluids to be loaded or unloaded at the same station without mixing._
+
+There are also some unconventional recipes, such as research, thermal power plant, and rocket silo.
+
+![research center module](figure/research_center_module.jpg)
+_Research center module, which doesn't need an output station_
+
+![thermal power plant module](figure/thermal_power_plant_module.jpg)
+_Thermal power plant with landfill to place on water._
+
+![rocket silo module](figure/rocket_silo_module.jpg)
+_Rocket silo module. Independent logistics robot network can be used within the module. The bottom station is for materials needed to build rockets, and the station on right is for payloads of the rocket, usually satellites._
+
+### Pros and cons of ROFL
+
+ROFL is optimized towards engineers, making it easy for players to divide and conquer production problems. When you need to implement a certain recipe, simply set up input and output stations in the grid that match the types of raw materials and products, and then design production lines between the stations. When the capacity of a recipe is insufficient, you can directly copy and paste the corresponding factory module and integrate it into the whole factory production without [plumbing up the inputs and outputs](https://alt-f4.blog/ALTF4-22/#the-goal-of-factorio).
```

Comment:
:D thanks for the sneaky plug here

Reply:
In fact, this is one of the reasons that motivated me to write this article, since I want to prove that blueprints can be even more hassle-free.